server/public/model/autotranslation.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"io"
+	"maps"
+)
+
+// TranslationType indicates the type of translated content
+type TranslationType string
+
+const (
+	TranslationTypeString TranslationType = "string"
+	TranslationTypeObject TranslationType = "object"
+)
+
+// TranslationState represents the state of a translation
+type TranslationState string
+
+const (
+	TranslationStateReady       TranslationState = "ready"
+	TranslationStateSkipped     TranslationState = "skipped"
+	TranslationStateUnavailable TranslationState = "unavailable"
+)
+
+// Translation represents a single translation result
+type Translation struct {
+	ObjectID   string           `json:"object_id"`
+	ObjectType string           `json:"object_type"`
+	Lang       string           `json:"lang"`
+	Provider   string           `json:"provider"`
+	Type       TranslationType  `json:"type"`
+	Text       string           `json:"text"`
+	ObjectJSON json.RawMessage  `json:"object_json,omitempty"`
+	Confidence *float64         `json:"confidence,omitempty"`
+	State      TranslationState `json:"state"`
+	Meta       map[string]any   `json:"meta,omitempty"`
+	NormHash   *string          `json:"norm_hash,omitempty"`
+}
+
+func (t *Translation) ToJSON() string {
+	b, _ := json.Marshal(t)
+	return string(b)
+}
+
+func (t *Translation) FromJSON(data io.Reader) (*Translation, error) {
+	err := json.NewDecoder(data).Decode(&t)
+	if err != nil {
+		return nil, err
+	}
+	return t, nil
+}
```

Comment:
We try to avoid `ToJSON` and `FromJSON` methods if they are not making any custom (un)marshaling. Can we remove them?